processor/memorylimiterprocessor/factory.go

```diff
@@ -99,15 +100,15 @@ func (f *factory) createLogsProcessor(
 
 // getMemoryLimiter checks if we have a cached memoryLimiter with a specific config,
 // otherwise initialize and add one to the store.
-func (f *factory) getMemoryLimiter(set processor.CreateSettings, cfg component.Config) (*memoryLimiter, error) {
+func (f *factory) getMemoryLimiter(set processor.CreateSettings, cfg component.Config) (*memoryLimiterProcessor, error) {
 	f.lock.Lock()
 	defer f.lock.Unlock()
 
 	if memLimiter, ok := f.memoryLimiters[cfg]; ok {
 		return memLimiter, nil
 	}
 
-	memLimiter, err := newMemoryLimiter(set, cfg.(*Config))
+	memLimiter, err := newMemoryLimiterProcessor(set, cfg.(*memorylimiter.Config))
```

Comment:
Use the same aliased type
```suggestion
	memLimiter, err := newMemoryLimiterProcessor(set, cfg.(*Config))
```

Reply:
i cant because golinter complains